lib/games_collection.rb

```diff
@@ -243,6 +243,40 @@ def worst_season(team_id)
     end
   end
 
+  def season_values(team_id, season)
+    values = {}
+    @games.each do |game|
+      if team_id == game.away_team_id || team_id == game.home_team_id
+        values[:win_percentage] = team_win_percentage(team_id, season)
+        values[:total_goals_scored] = total_home_goals(team_id) + total_away_goals(team_id)
+        values[:total_goals_against] = total_opponent_goals(team_id)
+        values[:average_goals_scored] = (average_away_score_of_team(team_id) + average_home_score_of_team(team_id)).round(2)
+        values[:average_goals_against] = average_goals_of_opponent(team_id)
+      end
+    end
+    values
+  end
+
+  def season_type(team_id, season)
+    type = {}
+    @games.each do |game|
+      if team_id == game.away_team_id || team_id == game.home_team_id
+      type[game.type.gsub(/\s+/, "_").downcase.intern] = season_values(team_id, season)
+      end
+    end
+    type
+  end
+
+  def seasonal_summary(team_id, season_id)
```

Comment:
As we saw from rspec, need to refactor to just take one argument (`team_id`) but we can keep working on it!